### PR TITLE
Task/remove group grants

### PR DESCRIPTION
**Pull request by esotrope**

### CODE REVIEW

File: common-web/src/main/java/org/opentestsystem/rdw/reporting/common/web/security/User.java [hunk 1]
```diff
@@ -64,15 +62,9 @@ public PermissionScope getPermissionScopeByPermissionId(final String id) {
         if (permission == null) {
             return PermissionScope.EMPTY;
         }
-
         return permission.getScope();
     }
 
-    public Map<Long, GroupGrant> getGroupsById() {
-        return groupsById;
-    }
```
Comment (esotrope) [inline common-web/src/main/java/org/opentestsystem/rdw/reporting/common/web/security/User.java]: removed groups from user object (the user is now primed for movement to the webapp module : j)

File: common/src/main/resources/common.sql.yml [hunk 2]
```diff
@@ -6,14 +6,6 @@ sql:
       where namespace=:namespace
       and language_code=:language_code
 
-  group:
-    membership:
-      findAllForUsername: >-
-        select sg.id, sg.subject_id, sg.school_id
-        from user_student_group usg
-          join student_group sg on sg.id = usg.student_group_id
-        where user_login=:user_login
```
Comment (esotrope) [inline common/src/main/resources/common.sql.yml]: we no longer prefetch the groups at login-time

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/model/GroupExamReportRequest.java [hunk 3]
```diff
@@ -5,11 +5,20 @@
...
         return groupGrant;
     }
 
+    public long getGroupId() {
+        // Bridge for existing reports.
+        if (groupGrant != null) {
+            return groupGrant.getId();
+        }
+        return groupId;
+    }
```
Comment (esotrope) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/model/GroupExamReportRequest.java]: I added a bridge here for existing reports until we talk about how we want to cutover.

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/model/jackson/AbstractExamReportRequestDeserializer.java [hunk 4]
```diff
@@ -48,10 +46,12 @@ private boolean hasNode(final JsonNode jsonNode, final String field) {
 
     private AbstractExamReportRequest deserializeLegacyJsonByProperty(final JsonNode jsonNode,
                                                                       final ObjectMapper mapper) throws IOException {
-        if (hasNode(jsonNode, "groupGrant")) {
+        // Bridge for old group requests
+        if (hasNode(jsonNode, "groupGrant")
+                || hasNode(jsonNode, "groupId")) {
```
Comment (esotrope) [inline report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/model/jackson/AbstractExamReportRequestDeserializer.java]: now checks for groupGrant or groupId

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/exam/group/GroupAssessmentSearchArgumentResolver.java [hunk 5]
```diff
@@ -28,7 +27,8 @@ public Object resolveArgument(
         final Map<String, String> pathVariables = getPathVariables(request);
         return GroupAssessmentSearch.builder()
                 .schoolYear(Integer.valueOf(request.getParameter("schoolYear")))
-                .groupGrant(getCurrentUser().getGroupsById().get(Long.valueOf(pathVariables.get("groupId"))))
```
Comment (esotrope) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/exam/group/GroupAssessmentSearchArgumentResolver.java]: i think we may be able to get rid of this `getCurrentUser()` method now actually

File: reporting-service/src/main/java/org/opentestsystem/rdw/reporting/exam/group/GroupAssessmentService.java [hunk 6]
```diff
@@ -3,6 +3,6 @@
 import org.opentestsystem.rdw.reporting.exam.AssessmentService;
 import org.springframework.security.access.prepost.PreAuthorize;
 
-@PreAuthorize("hasAuthority('PERM_GROUP_PII_READ') and #search.getGroupGrant() != null")
```
Comment (esotrope) [inline reporting-service/src/main/java/org/opentestsystem/rdw/reporting/exam/group/GroupAssessmentService.java]: no longer pre-checking group ownership

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/model/jackson/AbstractExamReportRequestDeserializer.java (hunk 4 above)
Comment (wtritch) [reply report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/model/jackson/AbstractExamReportRequestDeserializer.java]: Just want to double-check that you manually tested deserializing an old serialized request?

File: webapp/src/main/java/org/opentestsystem/rdw/reporting/security/auth/DefaultAuthorizationService.java [hunk 7]
```diff
@@ -165,15 +160,6 @@ public DefaultAuthorizationService(
...
     }
 
-    @Override
-    public Set<GroupGrant> getGroups(final Collection<Permission> permissions, final String username) {
-        // only lookup groups for those with the group PII permission
-        if (permissions.stream().noneMatch(permission -> permission.getId().equals(GroupPiiRead))) {
-            return ImmutableSet.of();
-        }
-        return ImmutableSet.copyOf(groupMembershipRepository.findAllForUsername(username));
-    }
```
Comment (wtritch) [inline webapp/src/main/java/org/opentestsystem/rdw/reporting/security/auth/DefaultAuthorizationService.java]: 👍

File: report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/model/jackson/AbstractExamReportRequestDeserializer.java (hunk 4 above)
Comment (esotrope) [reply report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/model/jackson/AbstractExamReportRequestDeserializer.java]: one of the unit tests asserts it already for me : j

**Pull request merged**